framework/exceptions/__init__.py

```diff
@@ -110,3 +110,6 @@ class TemplateHTTPError(HTTPError):
     def __init__(self, code, message=None, redirect_url=None, data=None, template=None):
         self.template = template
         super().__init__(code, message, redirect_url, data)
+
+class PendingVersionExists(FrameworkError):
+    pass
```

Comment:
Add `docstr` to explain what this error is and when it is raised. `PendingPreprintVersionExists` is a better name too.